coverage.py

```diff
@@ -509,6 +517,18 @@ def __init__(self, _db, api, datasource, workset_size=10,
             cutoff_time=cutoff_time
         )
 
-    def process_batch(self):
+    def process_batch(self, identifiers):
```

Comment:
This is the refactoring. I wanted to make sure that set_presentation_ready() is always called, so I implemented the most common version of process_batch(), moved the NotImplementedError to process_item(), and put the set_presentation_ready() call in handle_success (which can itself be overridden if you want to do something extra or _don't_ want to call set_presentation_ready()).